chia/wallet/wallet_state_manager.py

```diff
@@ -1428,6 +1429,15 @@ async def _add_coin_states(
                                     if derivation_record is None:  # not change
                                         to_puzzle_hash = coin.puzzle_hash
                                         amount += coin.amount
+                                        if coin_spend is not None:
+                                            # Check if this is a Clawback coin
+                                            puzzle: Program = coin_spend.puzzle_reveal.to_program()
+                                            solution: Program = coin_spend.solution.to_program()
+                                            uncurried = uncurry_puzzle(puzzle)
+                                            clawback_metadata = match_clawback_puzzle(uncurried, puzzle, solution)
+                                            if clawback_metadata is not None:
+                                                # Add the Clawback coin as the interested coin for the sender
+                                                await self.add_interested_coin_ids([coin.name()])
```

Comment:
since we already received all the data for this coin, do we really need to add it to interested coin ids here? can we reuse data we fetched to add it to the wallet?

Reply:
I thought about that. If it is an unspent coin, then we still need to add it to the table and that will trigger another coin state processing for the clawback coin. It doesn't compatible with the coin_added change we will do for the clawback.